crypto/hash/src/main/java/org/apache/shiro/crypto/hash/DefaultHashService.java

```diff
@@ -233,45 +233,6 @@ protected ByteSource getPublicSalt(HashRequest request) {
         return publicSalt;
     }
 
-    /**
-     * Combines the specified 'private' salt bytes with the specified additional extra bytes to use as the
-     * total salt during hash computation.  {@code privateSaltBytes} will be {@code null} }if no private salt has been
-     * configured.
-     *
-     * @param privateSalt the (possibly {@code null}) 'private' salt to combine with the specified extra bytes
-     * @param publicSalt  the extra bytes to use in addition to the given private salt.
-     * @return a combination of the specified private salt bytes and extra bytes that will be used as the total
-     *         salt during hash computation.
-     */
-    protected ByteSource combine(ByteSource privateSalt, ByteSource publicSalt) {
-
```

Comment:
One last thing, sorry it has taken so long to get to this.

This method signature needs remain the same for backwards compatibility, as well as the calling of this method on line `161`.

Just replace the body with a call to: `ByteSource.Util.combine()`

Thanks again!